package/kedro_viz/api/rest/responses.py

```diff
@@ -1,13 +1,37 @@
 """`kedro_viz.api.rest.responses` defines REST response types."""
 # pylint: disable=missing-class-docstring,too-few-public-methods,invalid-name
 import abc
+import logging
 from typing import Any, Dict, List, Optional, Union
 
+import fsspec
 import orjson
-from fastapi.responses import ORJSONResponse
+import packaging
+from fastapi.encoders import jsonable_encoder
+from fastapi.responses import JSONResponse, ORJSONResponse
+from kedro.io.core import get_protocol_and_path
 from pydantic import BaseModel
 
+try:
+    from importlib.metadata import version
+except ImportError:  # pragma: no cover
+    from importlib_metadata import version
+
 from kedro_viz.data_access import data_access_manager
+from kedro_viz.models.flowchart import (
+    DataNode,
+    DataNodeMetadata,
+    ParametersNodeMetadata,
+    TaskNode,
+    TaskNodeMetadata,
+    TranscodedDataNode,
+    TranscodedDataNodeMetadata,
+)
```

Comment:
Previously, in rest/router.py, we directly called models that were data classes, which is a common approach in [FastAPI](https://fastapi.tiangolo.com/advanced/dataclasses/).

Now, we've refactored this by moving the logic to two functions: `get_node_metadata_response` and `get_selected_pipeline_response`, which are in responses.py. This change is not new; it aligns with our existing approach for `get_default_response`, where the logic for /api/main resided in `responses.py`.

The reason we keep `get_default_response` in `responses.py` is because it's used in two places: in `router.py` when the user hits the endpoint `/api/main` and also during `kedro viz --save-file` in `server.py`, where we save these responses as JSON files. We've extended this approach to other API responses, such as node_metadata and selected_pipeline. This also fixed the previously broken` kedro viz --save-file` functionality, as we now save all the required files.

So yeah the above aligns with the same approach already used for  `get_default_response`. Happy to get on a call and discuss in more detail.

Reply:
We discussed this in person. Bottomline is that we should refactor Viz to use `dataclasses` in a better way and not do these response "conversions".